terraform-modules/github-runner/startup-script.sh

```diff
@@ -0,0 +1,127 @@
+#! /bin/bash
+
+set -e
+
+# Arguments from metadata
+ACTIONS_USER=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/actions-user -H "Metadata-Flavor: Google")
+VAULT_ADDR=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/vault-server -H "Metadata-Flavor: Google")
+ROLE_ID_PATH=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/role-id-path -H "Metadata-Flavor: Google")
+SECRET_ID_PATH=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/secret-id-path -H "Metadata-Flavor: Google")
+REPO=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/repo -H "Metadata-Flavor: Google")
+GITHUB_PAT_PATH=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/github-pat-secret-path -H "Metadata-Flavor: Google")
+RUNNER_LABELS=$(curl http://metadata.google.internal/computeMetadata/v1/instance/attributes/runner-labels -H "Metadata-Flavor: Google")
+RUNNER_NAME=$(curl "http://metadata.google.internal/computeMetadata/v1/instance/name" -H "Metadata-Flavor: Google")
+
+if [[ "$ROLE_ID_PATH" == gs* ]]; then
+    gsutil cp "$ROLE_ID_PATH" "$HOME/vault-agent/role-id"
+    ROLE_ID_PATH="$HOME/vault-agent/role-id"
+fi
+
+if [[ "$SECRET_ID_PATH" == gs* ]]; then
+    gsutil cp "$SECRET_ID_PATH" "$HOME/vault-agent/secret-id"
+    SECRET_ID_PATH="$HOME/vault-agent/secret-id"
+fi
```

Comment:
"If _PATH is a GCS path?"

Reply:
Pretty much. While the Terraform module validates that the _PATHs are `gs://...` paths, I actually found it useful for debugging the script to allow it to use local paths. Makes it a bit easier to run the script locally in a VM and it is also possible to attach disks to the compute instance that might have the IDs in them. Since I already had this I figured I wouldn't bother removing it since the TF module would safeguard against odd input and just leave this for debugging.